model/jpa/src/main/java/org/keycloak/events/jpa/JpaEventStoreProvider.java

```diff
@@ -146,7 +145,7 @@ public void close() {
 
     private EventEntity convertEvent(Event event) {
         EventEntity eventEntity = new EventEntity();
-        eventEntity.setId(UUID.randomUUID().toString());
+        eventEntity.setId(event.getId());
```

Comment:
What if `event.getId()` would be `null`? Could this be changed to
```suggestion
        eventEntity.setId(event.getId() == null ? UUID.randomUUID().toString() : event.getId());
```

Reply:
This comment applies to other places as well, could you fix those please?

Reply:
Fixed.